### PR TITLE
Add docs for spinners

### PR DESCRIPTION
**Pull request by glynnis**
Kinda mis-titled this branch, so please ignore the branch name.

This PR adds documentation for different spinner usage in Encore with buttons and action links, and improves the demo for spinners and buttons.

Addresses some of https://github.com/rackerlabs/encore-ui/issues/790, but I'm going to create an additional issue for "documenting other spinners across Encore" as I know there are others that are non-standard that we need to look into.

Please focus on wording and whether what I wrote for documentation makes sense outside of my designer brain! :+1: :ice_cream:

### CODE REVIEW

**Comment by glynnis** [comment]
New issue to address other non-standard spinners here: https://github.com/rackerlabs/encore-ui/issues/794

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/793/gallery.html)

File: demo/styleguide/buttons.html [hunk 1]
```diff
@@ -10,20 +10,27 @@ <h3 class="title clear">Customizing Buttons</h3>
...
-<p><span class="msg-info">ENCORE-SPECIFIC:</span> In the context of the Encore-UI, when is a button blue and when is a button green?</p>
-<ul class="list">
-    <li>If the form you are submitting results in CREATING or ADDING something, use green.</li>
-    <li>If the form you are submitting results in editing something or something auxiliary (like getting metadata from a different sources) keep it blue.</li>
-</ul>
+        <p><span class="msg-info">ENCORE-SPECIFIC:</span> In the context of the Encore-UI, when is a button blue and when is a button green?</p>
+        <ul class="list">
+            <li>If the form you are submitting results in CREATING or ADDING something, use green.</li>
+            <li>If the form you are submitting is within a modal, use green.</li>
+            <li>If the form you are submitting results in editing something or something auxiliary (like getting metadata from a different sources) keep it blue.</li>
```
Comment (parlarjb) [inline demo/styleguide/buttons.html]: Should "editing" be capitalized like you did with CREATING and ADDING?

Comment (glynnis) [inline demo/styleguide/buttons.html]: Done!

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/793/gallery.html)

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**